.github/workflows/publish-image.yml

```diff
@@ -28,13 +27,21 @@ jobs:
           registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Read build config YAML and read/set otelcol version
+        id: read-yaml
+        run: |
+          # Parse the YAML file and extract the otelcol version
+          OTELCOL_VERSION=$(yq -r '.dist.otelcol_version' otelcolarrow-build.yaml)
+          echo "OTELCOL_VERSION=${OTELCOL_VERSION}" >> $GITHUB_ENV
       # This step uses [docker/metadata-action](https://github.com/docker/metadata-action#about) to extract tags and labels that will be applied to the specified image. The `id` "meta" allows the output of this step to be referenced in a subsequent step. The `images` value provides the base name for the tags and labels.
       - name: Extract metadata (tags, labels) for Docker
         id: meta
         uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}-EXPERIMENTAL
           tags: |
+            # Set the OTELCOL_VERSION here
+            echo "::set-output name=otelcol-version-tag::${OTELCOL_VERSION}"
```

Comment:
I'm not sure this will work / do what we expect here... set-output is for setting the output for a step in github, which you've already done above. you'll instead want to do something like this 
```
type=raw,value=${OTELCOL_VERSION}
```